### PR TITLE
Fix passcode lth 5th

### PR DESCRIPTION
**Pull request by ningh1111**
Fix rotate issue.
https://jira.developers.mega.co.nz/browse/IOS-4622
https://jira.developers.mega.co.nz/browse/IOS-4623

Fix passcode length issue.
https://jira.developers.mega.co.nz/browse/IOS-4624

https://github.com/meganz/iOS_dev/pull/1262

### CODE REVIEW

File: LTHPasscodeViewController/LTHPasscodeViewController.m [hunk 1]
```diff
@@ -872,15 +873,17 @@ - (void)_setupOptionsButton {
...
+    [topBorder setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleBottomMargin];
     topBorder.frame = CGRectMake(0, 0, self.view.frame.size.width, 1);
-    topBorder.backgroundColor = _textFieldBorderColor.CGColor;
-    [_complexPasscodeOverlayView.layer addSublayer:topBorder];
+    topBorder.backgroundColor = _textFieldBorderColor;
+    [_complexPasscodeOverlayView addSubview:topBorder];
 
-    CALayer *bottomBorder = [CALayer layer];
+    UIView *bottomBorder = [UIView new];
+    [bottomBorder setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleBottomMargin];
```
Comment (simonwng) [inline LTHPasscodeViewController/LTHPasscodeViewController.m]: why don't we use autolayout?

Comment (simonwng) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: autoresizing mask is a very old technology which is replaced by autolayout

Comment (ningh1111) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: I tried to add top and bottom border to the UIView, but the method before does not work with autolayout. 
So I found this solution https://stackoverflow.com/a/29033559

Comment (simonwng) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: You need to set `translatesAutoresizingMaskIntoConstraints = false` for all the views to be added by autolayout.

Comment (ningh1111) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: updated

File: LTHPasscodeViewController/LTHPasscodeViewController.m [hunk 2]
```diff
@@ -1093,6 +1096,87 @@ - (void)updateViewConstraints {
                                     multiplier: 1.0f
                                       constant: 0.0f];
         [self.view addConstraints:@[overlayViewLeftConstraint, overlayViewY, overlayViewHeight, overlayViewWidth]];
+        
+        NSLayoutConstraint *topBorderWidth =
+        [NSLayoutConstraint constraintWithItem: _topBorder
```
Comment (simonwng) [inline LTHPasscodeViewController/LTHPasscodeViewController.m]: While you can use the old constraintWithItem method, but it is much easier to add constraints using layout anchor: https://developer.apple.com/documentation/uikit/nslayoutanchor?language=objc

like: _topBorder.widthAnchor constraints....

**Review by simonwng — APPROVED** [review]

File: LTHPasscodeViewController/LTHPasscodeViewController.m (hunk 2 above)
Comment (ningh1111) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: updated

File: LTHPasscodeViewController/LTHPasscodeViewController.m [hunk 3]
```diff
@@ -1093,6 +1096,16 @@ - (void)updateViewConstraints {
...
+        
+        [_topBorder.widthAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.widthAnchor].active = YES;
+        [_topBorder.centerXAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.centerXAnchor].active = YES;
+        [_topBorder.topAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.topAnchor].active = YES;
+        [_topBorder.heightAnchor constraintEqualToConstant:1.0f].active = YES;
+        
+        [_bottomBorder.widthAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.widthAnchor].active = YES;
+        [_bottomBorder.centerXAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.centerXAnchor].active = YES;
+        [_bottomBorder.topAnchor constraintEqualToAnchor:_complexPasscodeOverlayView.bottomAnchor constant:-1.0f].active = YES;
+        [_bottomBorder.heightAnchor constraintEqualToConstant:1.0f].active = YES;
```
Comment (simonwng) [inline LTHPasscodeViewController/LTHPasscodeViewController.m:1108]: nice

**Review by simonwng — APPROVED** [review]

**Pull request merged**